src/network/core/tcp_content.h

```diff
@@ -21,7 +21,7 @@
 /** Base socket handler for all Content TCP sockets */
 class NetworkContentSocketHandler : public NetworkTCPSocketHandler {
 protected:
-	void Close() override;
+	virtual void Close();
```

Comment:
I wonder if it is a good idea to have this `virtual`. `ClientNetworkContentSocketHandler::Close()` is now not called during dtor. No clue if that is an issue or not, so I am just wondering out loud.

Reply:
It already was virtual, by virtue of the empty implementation in NetworkSocketHandler/core.h.
Also, `ClientNetworkContentSocketHandler::Close()` was already not called from this destructor. Which is probably what you were expecting, but what is not the case and the whole point of this PR. As well as the LGTM rule I linked.

Reply:
Yes, I wrote why you made this PR :) Not what I am saying.

I am wondering if we should make a function that is being called from a dtor `virtual` at all. As it gives the illusion to the `override` will be called in all cases, which is not what is happening. In fact, I would argue, shouldn't it be private? Just to prevent any issues with this at all?

(again, I am just wondering out loud here)

Reply:
Making it private won't help, as the subclasses of this class call it. So at best it would be protected, as it already is.
Practically with the Python server component, this class is becoming somewhat pointless as well. And the destructor gets called when the application is stopped, so it not called `OnDisconnect()` is kind of pointless too. Anyway, I guess making it slightly more explicit won't hurt anyone.